src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,32 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.exceptions.RegisterException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_AMOUNT_CHARACTERS = 6;
+    private static final int MIN_AGE = 18;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user.getLogin().length() < MIN_AMOUNT_CHARACTERS) {
+            throw new RegisterException("User's login length = " + user.getLogin().length()
+                    + " but min length = " + MIN_AMOUNT_CHARACTERS);
+        } else if (user.getPassword().length() < MIN_AMOUNT_CHARACTERS) {
+            throw new RegisterException("User's password length = " + user.getPassword().length()
+                    + " but min length = " + MIN_AMOUNT_CHARACTERS);
+        } else if (user.getAge() < MIN_AGE || user.getAge() == null) {
+            throw new RegisterException("User's age " + user.getPassword().length()
+                    + " but allowed" + MIN_AGE);
+        }
+        for (User users:Storage.people) {
+            if (users.equals(user)) {
+                throw new RegisterException("User already registered!");
+            }
+        }
```

Comment:
Hard to read, try do not use (else if())
It is better to divide into methods and describe the necessary logic in them.
For example, you can create private methods:
private checkAge(User user) {
// age verification
}